### PR TITLE
Implement `axis_relative_direction` (`wl_pointer` version 9)

### PR DESCRIPTION
**Pull request by ids1024**
The relative direction is "inverted" for natural scrolling, and "identical" otherwise.

With this, `wl_seat` is now on the latest version.

### CODE REVIEW

File: src/backend/input/mod.rs [hunk 1]
```diff
@@ -357,6 +379,9 @@ pub trait PointerAxisEvent<B: InputBackend>: Event<B> {
 
     /// Source of the scroll event.
     fn source(&self) -> AxisSource;
+
+    /// Relative direction of phsical motion.
```
Comment (YaLTeR) [inline src/backend/input/mod.rs:383]:
```suggestion
    /// Relative direction of physical motion.
```

File: src/backend/libinput/mod.rs [hunk 2]
```diff
@@ -191,11 +193,21 @@ impl backend::PointerAxisEvent<LibinputInputBackend> for PointerScrollAxis {
...
+            Self::Wheel(_) => AxisSource::Wheel,
+            Self::Finger(_) => AxisSource::Finger,
+            Self::Continuous(_) => AxisSource::Continuous,
+        }
+    }
+
+    fn relative_direction(&self, axis: Axis) -> backend::AxisRelativeDirection {
+        let device = backend::Event::<LibinputInputBackend>::device(self);
+        match axis {
+            Axis::Vertical if device.config_scroll_natural_scroll_enabled() => {
```
Comment (YaLTeR) [inline src/backend/libinput/mod.rs:207]: Does it only apply to the vertical axis? The explanation of natural scrolling, "matching the motion of the scroll device with the direction of the content", does not read like it should be restricted to the vertical axis.

File: src/wayland/seat/pointer.rs [hunk 3]
```diff
@@ -305,9 +305,21 @@ where
...
+                        WlAxis::HorizontalScroll,
+                        details.relative_direction.0.into(),
+                    );
+                }
                 ptr.axis(details.time, WlAxis::HorizontalScroll, details.axis.0);
             }
             if details.axis.1 != 0.0 {
+                if ptr.version() >= 9 {
+                    ptr.axis_relative_direction(
+                        WlAxis::HorizontalScroll,
```
Comment (YaLTeR) [inline src/wayland/seat/pointer.rs:319]:
```suggestion
                        WlAxis::VerticalScroll,
```
well well well if it isn't the copy-paste mistake again

File: src/backend/libinput/mod.rs (hunk 2 above)
Comment (ids1024) [reply src/backend/libinput/mod.rs:207]: Seems you're right, when I test the natural scrolling option in libinput for either a touchpad or mouse.

Comment (ids1024) [reply src/backend/libinput/mod.rs:207]: The Kwin implementation in https://invent.kde.org/plasma/kwin/-/merge_requests/3456 has a comment `Maps to wl_pointer.axis_relative_direction. Currently used for y axis only`, but I'm not sure that's true in any was as implemented there...

File: anvil/src/input_handler.rs [hunk 4]
```diff
@@ -394,6 +394,7 @@ impl<BackendData: Backend> AnvilState<BackendData> {
         {
             let mut frame = AxisFrame::new(evt.time_msec()).source(evt.source());
             if horizontal_amount != 0.0 {
+                frame = frame.relative_direction(Axis::Horizontal, evt.relative_direction(Axis::Horizontal));
```
Comment (Drakulix) [inline anvil/src/input_handler.rs:397]: So should this now also be done for the vertical code path?

Comment (ids1024) [reply anvil/src/input_handler.rs:397]: Oh, I thought I had that here already. Added.

**Review by Drakulix — APPROVED** [review]

**Pull request merged**